docs/source/user_guide/pipeline-components.md

```diff
@@ -50,18 +50,186 @@ Component details and demo pipelines can be found in the `https://github.com/ely
+
+```bash
+$ elyra-metadata install component-registries \
+       --name="filter_components" \
+       --display_name="filter components" \
+       --description="filter text in files" \
+       --runtime=kfp \
+       --location_type=URL \
+       --paths="['https://raw.githubusercontent.com/elyra-ai/elyra/master/etc/config/components/kfp/filter_text_using_shell_and_grep.yaml']" \
+       --categories='["file operations"]' \
+       --replace
+```
+
+Note: You must specify all property values, not only the ones that you want to modify.
+
+Refer to section [Configuration properties](#configuration-properties) for a parameter description.
+
+#### Deleting components from the registry
+
+To delete a component registry entry and its component definitions:
+
+```bash
+$ elyra-metadata remove component-registries \
+       --name="filter_components"
+```
+
+Refer to section [Configuration properties](#configuration-properties) for a parameter description.
+
+### Configuration properties
+
+The component registry entry properties are defined as follows. The string in the headings below, which is enclosed in parentheses, denotes the CLI option name.
+
+##### Name (display_name)
+
+A user-friendly name for the registry entry. Note that the registry entry name is not displayed in the palette. This property is required.
+
+Example: `data load components`
+
+##### N/A (name)
+
+A unique identifier for this registry entry. A value is generated from `Name` if no value is provided.
+
+Example: `data_load_components`
+
+##### Description (description)
+
+A description for the registry entry.
+
+Example: `Load data from external data sources`
+
+##### Category (categories)
+
+In the pipeline editor palette components are grouped into categories to make them more easily accessible. If no category is provided, the components defined by this registry entry are added to the palette under `no category`.  
+
+Examples (CLI):
+
+- `['load data from database']`
+- `['train model','pytorch']`
+
+
+##### Runtime (runtime)
+
+The runtime environment that supports the component(s). Valid values are `kfp` and `airflow`. This property is required.
+
+Example:
+
+- `airflow`
+
+##### Location Type (location_type)
+
+The location type identifies the format that the value(s) provided in `Paths` represent. Supported types are `URL`, `Filename`, or `Directory`. This property is required.
+
+- `URL`: The provided `Paths` identify web resources. The pipeline editor loads the specified URLs using anonymous HTTP `GET` requests.
+- `Filename`: The provided `Paths` identify files in the file system where JupyterLab/Elyra is running. `~` may be used to denote the user's home directory.
+
+The `Directory` location type is used to _add components during the Elyra build process_. The provided `Paths` must identify existing directories in [`etc/config/components`](https://github.com/elyra-ai/elyra/tree/master/etc/config/components).
```

Comment:
I don't think the _build-time_ aspect of this is correct.  `Directory` should be a location where users can dump their component specifications and independent of the "build process".  That directory can reside anywhere on the system (provided users have appropriate access rights).  If I'm understanding things, we should keep `etc/config/components` to ourselves (i.e., it is reserved for the _build-time_ needs we have to deliver _factory assets_) and the only "non-absolute" paths allowed are the two from the factory: `airflow` and `kfp`.  (Again, if I'm understanding things correctly.)

`Directory` should then be another bullet in the list and _just an option_ like `URL` and `Filename` - there's nothing special about it other than it implies all files in its location are component definitions.

Reply:
Changed and removed references to relative path [for now].